integration-tests/src/test_loop/tests/validator_selection.rs

```diff
@@ -0,0 +1,128 @@
+    let epoch_length = 10;
+
+    let accounts =
+        (0..6).map(|i| format!("account{}", i).parse().unwrap()).collect::<Vec<AccountId>>();
+    let clients = accounts.iter().cloned().collect_vec();
+    let validators = vec![
+        AccountInfo {
+            account_id: accounts[0].clone(),
+            public_key: create_test_signer(accounts[0].as_str()).public_key(),
+            amount: 50_000 * 62_500 * ONE_NEAR,
+        },
+        AccountInfo {
+            account_id: accounts[1].clone(),
+            public_key: create_test_signer(accounts[1].as_str()).public_key(),
+            amount: 50_000 * 62_500 * ONE_NEAR,
+        },
+        AccountInfo {
+            account_id: accounts[2].clone(),
+            public_key: create_test_signer(accounts[2].as_str()).public_key(),
+            amount: 10_000 * ONE_NEAR,
+        },
+    ];
+    let mut genesis_builder = TestGenesisBuilder::new();
+    genesis_builder
+        .protocol_version(protocol_version)
+        .genesis_time_from_clock(&test_loop_builder.clock())
+        .shard_layout(epoch_config_store.get_config(protocol_version).shard_layout.clone())
+        .epoch_length(epoch_length)
+        .validators_raw(validators, 3, 3, 3)
+        .max_inflation_rate(Rational32::new(0, 1));
+    let initial_balance = 1_000_000 * ONE_NEAR;
+    for account in &accounts {
+        genesis_builder.add_user_account_simple(account.clone(), initial_balance);
+    }
+    let (genesis, _) = genesis_builder.build();
+
+    let TestLoopEnv { mut test_loop, datas: node_data, tempdir } = test_loop_builder
+        .genesis(genesis)
+        .epoch_config_store(epoch_config_store)
+        .clients(clients)
+        .build();
+
+    let sender = node_data[0].client_sender.clone();
+    let handle = sender.actor_handle();
+    let client = &test_loop.data.get(&handle).client;
+
+    let epoch_id = client
+        .epoch_manager
+        .get_epoch_id_from_prev_block(&client.chain.head().unwrap().last_block_hash)
+        .unwrap();
+    let epoch_info = client.epoch_manager.get_epoch_info(&epoch_id).unwrap();
+    let protocol_version = client.epoch_manager.get_epoch_protocol_version(&epoch_id).unwrap();
+    let validators = get_epoch_all_validators(client);
+    let total_stake = validators
+        .iter()
+        .map(|v| {
+            let account_id = &v.parse().unwrap();
+            epoch_info.get_validator_stake(account_id).unwrap()
+        })
+        .sum::<u128>();
+    // sanity checks
+    assert!(protocol_version < ProtocolFeature::FixStakingThreshold.protocol_version());
+    assert_eq!(validators.len(), 2);
+    assert_eq!(total_stake / ONE_NEAR, 6_250_000_000);
+    // prior to threshold fix
+    // threshold = min_stake_ratio * total_stake
+    //           = (1 / 62_500) * 6_250_000_000
+    // FIXME eagr This is failing. What is the expected validator threshold?
+    assert_eq!(epoch_info.seat_price() / ONE_NEAR, 100_000);
```

Comment:
```
thread 'test_loop::tests::validator_selection::slow_test_validator_selection' panicked at integration-tests/src/test_loop/tests/validator_selection.rs:92:5:
assertion `left == right` failed
  left: 16666
 right: 100000
```
The result is 1/6 of what I expect it to be. What am I missing here?

Reply:
Oh right :(
This is because of 6 shards and this very well hidden artifact https://github.com/near/nearcore/blob/8756ac530/chain/epoch-manager/src/validator_selection.rs#L370-L383

Actually, it would be even better to make a protocol feature to get rid of this ratio (in other words, always set it to 1). It is no longer needed because chunk producers are no longer the lightest nodes.

If you would like to, you could fix this as well. Or just get around it with forcing only 1 chunk producer in epoch configs with very big stake...

Reply:
done :) what else needs to be done to stablize the feature?

Reply:
> make a protocol feature

Oh, I missed out on this part. Guess we should do it in a separate PR.